nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/HealthCheckService.java

```diff
@@ -300,7 +304,8 @@ public void shutdownHealthCheck() {
     }
 
     public BaseHealthCheck getCheck(String serviceName) {
-        return registeredTasks.get(serviceName).getCheck();
+        HealthCheckTask task = registeredTasks.get(serviceName);
+        return task == null ? null : task.getCheck();
```

Comment:
Here NPE occurred in test scenarios if there was no task for the `serviceName`. Returning `null` allows the caller to take care of the case.